cogs/annual_handover_and_reset.py

```diff
@@ -222,3 +227,109 @@ async def annual_roles_reset(self, ctx: TeXBotApplicationContext) -> None:
 
         logger.debug("Execution of reset roles command complete!")
         await initial_response.edit(content=":white_check_mark: Role reset complete!")
+
+class IterateYearChannelsCommandCog(TeXBotBaseCog):
+    """Cog class that defines the "/iterate-year-channels" command."""
+
+    @discord.slash_command(  # type: ignore[no-untyped-call, misc]
+        name="iterate-year-channels",
+        description="Iterates the year channels, archiving and creating channels as needed",
+    )
+    async def iterate_year_channels(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition and callback response of the "iterate-year-channels" command.
+
+        This command:
+        - Archives the current final-years channel
+        - Renames the current second year channel to final-years
+        - Renames the current first year channel to second-years
+        - Creates a new first years channel
+        """
+        main_guild: discord.Guild = self.bot.main_guild
+        guest_role: discord.Role = await self.bot.guest_role
+
+        initial_message: discord.Interaction | discord.WebhookMessage = await ctx.respond(
+            content=":hourglass: Processing year channel iteration... :hourglass:",
+        )
+
+        final_year_channel: discord.TextChannel | None = discord.utils.get(
+            main_guild.text_channels,
+            name="final-years",
+        )
+
+        if final_year_channel:
+            await initial_message.edit(
+                content=":hourglass: Archiving final year channel... :hourglass:",
+            )
+            archivist_role: discord.Role = await self.bot.archivist_role
+
+            await final_year_channel.set_permissions(guest_role, overwrite=None)
+            await final_year_channel.set_permissions(archivist_role, read_messages=True)
+
+            await final_year_channel.edit(
+                name="final-years-" + str(datetime.datetime.now(tz=datetime.UTC).year),
```

Comment:
Use f-string instead of string concatenation (more Pythonic): `name=f"final-years-{datetime.datetime.now(tz=datetime.UTC).year}",`